src/main/java/de/plushnikov/intellij/plugin/processor/clazz/log/CustomLogProcessor.java

```diff
@@ -0,0 +1,155 @@
+package de.plushnikov.intellij.plugin.processor.clazz.log;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiClass;
+import de.plushnikov.intellij.plugin.lombokconfig.ConfigDiscovery;
+import de.plushnikov.intellij.plugin.lombokconfig.ConfigKey;
+import de.plushnikov.intellij.plugin.problem.ProblemBuilder;
+import de.plushnikov.intellij.plugin.util.PsiAnnotationUtil;
+import lombok.CustomLog;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * @author Adam Juraszek
+ */
+public class CustomLogProcessor extends AbstractLogProcessor {
+  // same as in lombok.core.configuration.LogDeclaration
+  private static final Pattern PARAMETERS_PATTERN = Pattern.compile("(?:\\(([A-Z,]*)\\))");
+  private static final Pattern DECLARATION_PATTERN =
+    Pattern.compile("^(?:([^ ]+) )?([^(]+)\\.([^(]+)(" + PARAMETERS_PATTERN.pattern() + "+)$");
+
+
+  private static class LoggerInitializerDeclaration {
+    final List<LoggerInitializerParameter> withTopic;
+    final List<LoggerInitializerParameter> withoutTopic;
+
+    LoggerInitializerDeclaration(List<LoggerInitializerParameter> withTopic, List<LoggerInitializerParameter> withoutTopic) {
+      this.withTopic = withTopic;
+      this.withoutTopic = withoutTopic;
+    }
+  }
+
+  public CustomLogProcessor(@NotNull ConfigDiscovery configDiscovery) {
+    super(configDiscovery, CustomLog.class);
+  }
+
+  @NotNull
+  private static String getCustomDeclaration(@NotNull PsiClass psiClass) {
+    return ConfigDiscovery.getInstance().getStringLombokConfigProperty(ConfigKey.LOG_CUSTOM_DECLARATION, psiClass);
+  }
+
+  @NotNull
+  @Override
+  public String getLoggerType(@NotNull PsiClass psiClass) {
+    final Matcher declarationMatcher = DECLARATION_PATTERN.matcher(getCustomDeclaration(psiClass));
+    if (!declarationMatcher.matches()) {
+      return "!@#$%^&*()"; // unfortunately this is not guarded by validation; we must return something to be tested for equality
+    }
+    String loggerType = declarationMatcher.group(1);
+    if (loggerType == null) {
+      loggerType = declarationMatcher.group(2);
+    }
+    return loggerType;
+  }
+
+  @NotNull
+  @Override
+  String getLoggerInitializer(@NotNull PsiClass psiClass) {
+    final Matcher declarationMatcher = DECLARATION_PATTERN.matcher(getCustomDeclaration(psiClass));
```

Comment:
Can you move all of matcher/parse logic to one place and add some unit test for this?